sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -167,8 +167,10 @@ private[hive] trait HiveStrategies {
 
   object Scripts extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
-      case logical.ScriptTransformation(input, script, output, child) =>
-        ScriptTransformation(input, script, output, planLater(child))(hiveContext) :: Nil
+      case logical.ScriptTransformation(input, script, output, child, schema) =>
+          ScriptTransformation(input, script, output,
+            planLater(child), schema.map{ case s: HiveScriptIOSchema => s }.get
```

Comment:
Given that it is safe to just call `get` here it seems like `Option` might not be the correct choice for the data model.  I'd also do the casting by place a type annotation in the match.

```scala
case logical.ScriptTransformation(input, script, output, child, schema: HiveScriptIOSchema) =>
```